### PR TITLE
#201 - Allow -1 for :lazy_inputs_interval option

### LINKED ISSUE #201: Allow -1 for :lazy_inputs_interval option
The meaning is: never make the set explicit.

### PR DESCRIPTION
**Pull request by schillic**
Closes #201.

### CODE REVIEW

**Comment by schillic** [comment]
The solution is more a hack because in Julia `k % -1 == 0`.
An alternative could be to internally store a function that takes a number and returns `true` or `false`.

File: src/options.jl [hunk 1]
```diff
@@ -590,6 +592,35 @@ function compute_blocks(vars, partition)
...
+The default value is ``0``.
+If the input is ``-1``, we set the output to ``2147483647`` (maximum `Int32`
+value).
+"""
+function check_and_add_lazy_inputs_interval!(dict::Dict{Symbol,Any},
+                                             dict_copy::Dict{Symbol,Any})
+    check_aliases!(dict, dict_copy, [:lazy_inputs_interval])
+    if haskey(dict_copy, :lazy_inputs_interval)
+        if dict_copy[:lazy_inputs_interval] == -1
+            dict_copy[:lazy_inputs_interval] = 2147483647 # very big number
```
Comment (mforets) [inline src/options.jl]: for a time horizon of 100, 2147483647 corresponds to a time step 4.656612875245797e-8

for the same time horizon, 9223372036854775807 (obtained with `typemax(Int)` in my machine) is a time step 1.0842021724855044e-17  => maybe take this number, `typemax(Int64)`

Comment (schillic) [reply src/options.jl]: I do not think that 2147483647 is a realistic step number, no matter how small the step size is. I do not want to require `Int64` in this project for such a tiny thing.
I can offer to add the proposed function solution instead, which might be clearer and more efficient. But first we should merge #199 which conflicts with that.

**Comment by schillic** [comment]
I added the new changes to #199.

**Pull request closed without merging**